### ISSUE TITLE
Min/Max dependency  testing should exercise optional requirements

### ISSUE THREAD

**Issue by scbedd** [Azure/azure-sdk-for-python#14789]
This issue is springing from discussion on [this PR](https://github.com/Azure/azure-sdk-for-python/pull/14786 )

Right now this includes:

- `aiohttp`

I'm not currently aware of any other "optional" packages like this.

However, before we can reliably do that, we need to establish a contract on how we can DEFINE the min and max of this package. Right now we assume ANY version should work appropriately.

**Comment by scbedd** [comment]
Closing in favor of #5189

Progress visible in [feature/optional-install](https://github.com/scbedd/azure-sdk-for-python/tree/feature/optional-install)
